mainapp/helpers.py

```diff
@@ -29,8 +31,8 @@
 with open(sim_books_path, 'rb') as handle:
     sim_books_dict = pickle.load(handle)
 
-
 cols = ['original_title', 'authors', 'average_rating', 'image_url', 'book_id']
+priority_list = ['fiction', 'fantasy', 'classics', 'contemporary', 'mystery', 'nonfiction', 'paranormal', 'romance', 'history', 'thriller', 'horror', 'memoir', 'comics', 'biography', 'philosophy', 'science', 'crime', 'psychology', 'christian', 'business', 'poetry', 'music', 'religion', 'manga', 'art', 'spirituality', 'cookbooks', 'travel', 'ebooks', 'sports', 'suspense']
```

Comment:
Why is this needed?

Reply:
okey, so first off we loop through all the book ids that were given as input and append genres to a list.
that list looks something like this 
```python
['art','art','mystery','romance','romance','fiction','fiction']
```
when this list is passed on to Counter(), it returns something like
```python
{'art':2, 'romance':2, 'fiction':2, 'mystery':1}
```
and the Counter(list).most_common(1) returns the very first item from the dictionary
```
[{'art',2)]
```
but because of this in some cases the order of genre changes after Counter(), resulting in test genre and recommended genre being different, failing the test checks.
So after passing the list to Counter()
```python
{'art':2, 'romance':2, 'fiction':2, 'mystery':1}
```
the genre with highest priority, i.e. the greatest book count is recommended. and `priority_list` stores that order from 0 to n in desending order by

1. Slice out the highest value pairs from the dictionary
```python
most_common_dict = {u: v for u, v in genre_count.items() if v == max_value}
# output {'art':2, 'romance':2, 'fiction':2}
```
2. map out the priority list with index values
```python
index_map = {v: i for i, v in enumerate(priority_list)}
# output {'fiction':0, 'fantasy':1, 'classics':2, 'contemporary':3, ........}
```
3. sorting the dictionary pairs on the index value of a genre
```python
final_list = sorted(most_common_dict.items(), key=lambda pair: index_map[pair[0]])
# output [('fiction',2), ('romance',2), ('art',2)]
```
4. return the highest priority genre
```python
most_common_genre = final_list[0][0]
# output 'fiction'
```

Reply:
I don't think it is a good practise to hardcode the priority list, instead what you can do to remove the conflict is order the conflict same count genre according to their alphabetical order(take the first one) or according to the count of that genre in the database(take the max one).

Reply:
I have removed the hard coded list and I am implementing a better way to do the same, will make a pr by EOD